tests/core/v5_1/test_v51_rpc_handlers.py

```diff
@@ -73,8 +100,99 @@ def ping_param_w3(request, alice, bob, bob_network):
+    distances = set()
+
+    for _ in range(10):
+        enr = ENRFactory()
+        distances.add(compute_log_distance(bob.node_id, enr.node_id))
+        bob.enr_db.set_enr(enr)
+
+    # request with positional single distance
+    enrs_at_0 = await make_request("discv5_findNodes", [bob_node_id_param, 0])
+    assert all(isinstance(ENR.from_repr(enr), ENR) for enr in enrs_at_0)
+
+    # request with multiple distances
+    enrs_at_some_distance = await make_request(
+        "discv5_findNodes", [bob_node_id_param, tuple(distances)],
+    )
+    assert all(isinstance(ENR.from_repr(enr), ENR) for enr in enrs_at_some_distance)
+
+
+@pytest.mark.trio
+async def test_v51_rpc_findNodes_invalid_params(
+    make_request, invalid_node_id, alice, bob
+):
+    # bad node_id
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [invalid_node_id, 0])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [invalid_node_id, [0]])
+
+    # invalid distances
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), -1])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), 257])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), 1.2])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), []])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), "1"])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), [1, "2"]])
+
+    # wrong params count
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex()])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_findNodes", [bob.node_id.hex(), 0, "extra"])
+
+
+@pytest.mark.trio
+async def test_v51_rpc_findNodes_w3(make_request, bob_node_id_param, alice, bob, w3):
+    distances = set()
+
+    for _ in range(10):
+        enr = ENRFactory()
+        distances.add(compute_log_distance(bob.node_id, enr.node_id))
+        bob.enr_db.set_enr(enr)
+
+    # request with positional single distance
+    enrs_at_0 = await trio.to_thread.run_sync(
+        w3.discv5.find_nodes, bob_node_id_param, 0
+    )
+    assert all(isinstance(enr, ENR) for enr in enrs_at_0)
+
+    # request with multiple distances
+    enrs_at_some_distance = await trio.to_thread.run_sync(
+        w3.discv5.find_nodes, bob_node_id_param, tuple(distances),
+    )
+    assert all(isinstance(ENR.from_repr(enr), ENR) for enr in enrs_at_some_distance)
```

Comment:
`from_repr` will always return an `ENR`, I think the `isinstance` is extraneous.

I'm also surprised that both of these tests pass! If line 192 passes then `find_nodes` returns straight ENRs, if this line passes then `find_nodes` returns strings. Is it possible that one of these tests is not testing anything, because the result set is an empty list?